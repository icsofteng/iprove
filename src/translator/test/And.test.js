const {translate: translate_z3} = require('../../translator/z3')
const {translate: translate_latex} = require('../../translator/latex')

const test_constants = [{ value: 'P', varType: 'Bool' }, { value: 'Q', varType: 'Bool' }]
const test_rules = [
  {
    type: 'binary',
    symbol: 'and',
    lhs: {
      type: 'literal',
      value: 'P'
    },
    rhs: {
      type: 'literal',
      value: 'Q'
    },
  },
  { 
    type: 'literal',
    value: 'P'
  }
]

test('And test z3', () => {
  expect(translate_z3(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})

test('And test latex', () => {
  expect(translate_latex(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})