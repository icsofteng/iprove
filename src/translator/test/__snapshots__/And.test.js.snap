// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`And test latex 1`] = `
Array [
  "P \\\\land Q",
  "P",
]
`;

exports[`And test z3 1`] = `
"(declare-const P Bool)
(declare-const Q Bool)
(assert (and P Q))
(assert (not P))
(check-sat)"
`;
